daskms/experimental/utils.py

```diff
@@ -31,14 +35,8 @@ def column_iterator(variables, columns):
     if columns is None or columns == "ALL":
         return variables.items()
     else:
-        column_set = set(columns)
-        unknown_columns = column_set - set(variables.keys())
-
-        if len(unknown_columns) > 0:
-            raise ValueError(f"{unknown_columns} are not present "
-                             f"in the dataset.")
-
-        return ((k, variables[k]) for k in column_set)
+        columns = set(columns) & set(variables.keys(0))
```

Comment:
This line gives me
```
Traceback (most recent call last):
  File "/home/landman/venvs/pfbenv3.7/bin/pfbworkers", line 11, in <module>
    load_entry_point('pfb-clean', 'console_scripts', 'pfbworkers')()
  File "/home/landman/venvs/pfbenv3.7/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/landman/venvs/pfbenv3.7/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/landman/venvs/pfbenv3.7/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/landman/venvs/pfbenv3.7/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/landman/venvs/pfbenv3.7/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/landman/Projects/pfb-clean/pfb/workers/grid/predict.py", line 89, in predict
    return _predict(ms, stack, **kw)
  File "/home/landman/Projects/pfb-clean/pfb/workers/grid/predict.py", line 174, in _predict
    ms, nband=nband)
  File "/home/landman/Projects/pfb-clean/pfb/utils/misc.py", line 224, in chan_to_band_mapping
    xds = xds_from_ms(ims, chunks={"row": -1}, columns=('TIME'))
  File "/home/landman/Projects/dask-ms/daskms/dask_ms.py", line 332, in xds_from_storage_ms
    return xds_from_zarr(store, **kwargs)
  File "/home/landman/Projects/dask-ms/daskms/experimental/zarr/__init__.py", line 322, in xds_from_zarr
    for name, zarray in column_iterator(group, columns):
  File "/home/landman/Projects/dask-ms/daskms/experimental/utils.py", line 38, in column_iterator
    columns = set(columns) & set(variables.keys(0))
TypeError: keys() takes 1 positional argument but 2 were given
```

Reply:
Thanks, this was a typo and should now be corrected.